### DISCUSSION TITLE
watch(route.query / route.params not triggering

### DISCUSSION THREAD

**Discussion by u007** [nuxt/nuxt#16200] [Questions] [answered]
hi,

when i was using vue3, this works. but once i ported to my nuxt3 project, using all stable release version, it does not trigger.
i even tried deep: true, still does not trigger:

i trigger this code below
```
//....
const $router = useRouter()
$router.push({ path: $route.path, query: { ...$route.query, page: p, pageSize pageSize.value } })
```

and watch inside a composable

```
const $route = useRoute()

 watch($route.query, (v) => {
    console.log('watch!!!', v)
  }, { deep: true, immediate: true })
```

update: i realize this works , but what is the different? why do we need to wrap it as callback?

```
const $route = useRoute()

 watch(() => $route.query, (v) => {
    console.log('watch!!!', v)
  }, { deep: true, immediate: true })
```

**Comment by wobsoriano** [comment, marked as answer]
You can't watch a property of a reactive object.

https://vuejs.org/guide/essentials/watchers.html#basic-example
